src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-overview-dashboard/rgw-overview-dashboard.component.ts

```diff
@@ -0,0 +1,184 @@
+import {
+  FeatureTogglesMap$,
+  FeatureTogglesService
+} from '~/app/shared/services/feature-toggles.service';
+import { RefreshIntervalService } from '~/app/shared/services/refresh-interval.service';
+
+import { RgwDaemonService } from '~/app/shared/api/rgw-daemon.service';
+import { RgwRealmService } from '~/app/shared/api/rgw-realm.service';
+import { RgwZoneService } from '~/app/shared/api/rgw-zone.service';
+import { RgwZonegroupService } from '~/app/shared/api/rgw-zonegroup.service';
+import { RgwBucketService } from '~/app/shared/api/rgw-bucket.service';
+import { RgwUserService } from '~/app/shared/api/rgw-user.service';
+import { PrometheusService } from '~/app/shared/api/prometheus.service';
+import { RgwPromqls as queries } from '~/app/shared/enum/dashboard-promqls.enum';
+import moment from 'moment';
+
+@Component({
+  selector: 'cd-rgw-overview-dashboard',
+  templateUrl: './rgw-overview-dashboard.component.html',
+  styleUrls: ['./rgw-overview-dashboard.component.scss']
+})
+export class RgwOverviewDashboardComponent implements OnInit, OnDestroy {
+  interval = new Subscription();
+  permissions: Permissions;
+  enabledFeature$: FeatureTogglesMap$;
+  rgwDaemonCount = 0;
+  rgwRealmCount = 0;
+  rgwZonegroupCount = 0;
+  rgwZoneCount = 0;
+  rgwBucketCount = 0;
+  objectCount = 0;
+  UserCount = 0;
+  totalPoolUsedBytes = 0;
+  queriesResults: any = {
+    RGW_REQUEST_PER_SECOND: '',
+    BANDWIDTH: '',
+    AVG_GET_LATENCY: '',
+    AVG_PUT_LATENCY: ''
+  };
+  timerGetPrometheusDataSub: Subscription;
+  daemonSub: Subscription;
+  realmSub: Subscription;
+  ZonegroupSub: Subscription;
+  ZoneSUb: Subscription;
+  UserSub: Subscription;
+  HealthSub: Subscription;
+  BucketSub: Subscription;
+  timerTime = 30000;
+  readonly lastHourDateObject = {
+    start: moment().unix() - 3600,
+    end: moment().unix(),
+    step: 12
+  };
+  rgwDaemonNames: any = [];
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    private featureToggles: FeatureTogglesService,
+    private healthService: HealthService,
+    private refreshIntervalService: RefreshIntervalService,
+    private rgwDaemonService: RgwDaemonService,
+    private rgwRealmService: RgwRealmService,
+    private rgwZonegroupService: RgwZonegroupService,
+    private rgwZoneService: RgwZoneService,
+    private rgwBucketService: RgwBucketService,
+    private rgwUserService: RgwUserService,
+    public prometheusService: PrometheusService
+  ) {
+    this.permissions = this.authStorageService.getPermissions();
+    this.enabledFeature$ = this.featureToggles.get();
+  }
+
+  ngOnInit() {
+    this.interval = this.refreshIntervalService.intervalData$.subscribe(() => {
+      this.daemonSub = this.rgwDaemonService.list().subscribe((data: any) => {
+        this.rgwDaemonNames = data.map((daemon: any) => {
+          return daemon['id'];
+        });
+        this.rgwDaemonCount = data.length;
+      });
```

Comment:
I'll repeat my recommendation for handling subscriptions and async code in Angular. Let's just save the observables in the TS code, but don't subscribe to those. That can be done in the HTML side via `async` pipe, which automatically handles unsubscriptions for us and results in cleaner code:
```html
    <cd-card *ngIf="daemons$ | async as daemons" cardTitle="Daemons"
             i18n-title
             link="/rgw/daemons"
             class="col-sm-2 px-3 d-flex"
             aria-label="Daemons card">
      <span class="ms-4 me-4">
        <h1>{{ daemons.length }}</h1>
      </span>
    </cd-card>

Reply:
Tried this approach..since we have a short interval after which the api calls are repeated, the async pipe is causing some flickering in the UI.